scrapy_po/utils.py

```diff
@@ -129,3 +133,19 @@ def build_instances(plan: andi.Plan, providers):
         else:
             instances[cls] = cls(**kwargs_spec.kwargs(instances))
     raise returnValue(instances)
+
+
+def callback_for(page_cls):
+    """Helper for defining callbacks for pages with to_item methods."""
+
+    # When the callback is used as an instance method of the spider, it expects
+    # to receive 'self' as its first argument. When used as a simple inline
+    # function, it expects to receive a response as its first argument.
+    #
+    # To avoid a TypeError, we need to receive a list of unnamed arguments and
+    # a dict of named arguments after our injectable.
+    def parse(*args, page: page_cls, **kwargs):
```

Comment:
I've had to revert the `*args` and to include a `**kwargs` to make sure this parse method is able to receive optional 'self' instances as arguments. It's well explained in the comment above.